src/dali/plugin/input/rest/binance_us.py

```diff
@@ -0,0 +1,1329 @@
+_OBTAIN_AMOUNT: str = "obtainAmount"
+_ORDER: str = "order"  # CCXT only variable
+_ORDER_NO: str = "orderNo"
+_PAGE_INDEX: str = "pageIndex"
+_PAGE_SIZE: str = "pageSize"
+_POSITION_ID: str = "positionId"
+_PRICE: str = "price"
+_PRODUCT: str = "product"
+_PROFIT_AMOUNT: str = "profitAmount"
+_REDEMPTION: str = "REDEMPTION"
+_ROWS: str = "rows"
+_SELL: str = "sell"  # CCXT only variable
+_SIDE: str = "side"  # CCXT only variable
+_SIZE: str = "size"
+_STAKING: str = "STAKING"
+_START_TIME: str = "startTime"
+_STATUS: str = "status"
+_SOURCE_AMOUNT: str = "sourceAmount"
+_SUBSCRIPTION: str = "SUBSCRIPTION"
+_SYMBOL: str = "symbol"
+_TIME: str = "time"
+_TIMESTAMP: str = "timestamp"  # CCXT only variable
+_TRAN_ID: str = "tranId"
+_TRANSACTION_TYPE: str = "transactionType"
+_TOTAL: str = "total"
+_TOTAL_FEE: str = "totalFee"
+_TOTAL_NUM: str = "totalNum"
+_TYPE: str = "type"
+_TX_ID: str = "txid"  # CCXT doesn't capitalize I
+_TXN_TYPE: str = "txnType"
+_UPDATE_TIME: str = "updateTime"
+_USERNAME: str = "userName"
+_WITHDRAWAL: str = "withdrawal"  # CCXT only variable
+
+# Time period constants
+_NINETY_DAYS_IN_MS: int = 7776000000
+_THIRTY_DAYS_IN_MS: int = 2592000000
+_ONE_DAY_IN_MS: int = 86400000
+_MS_IN_SECOND: int = 1000
+
+# Record limits
+_DEPOSIT_RECORD_LIMIT: int = 1000
+_DIVIDEND_RECORD_LIMIT: int = 500
+_DUST_TRADE_RECORD_LIMIT: int = 100
+_INTEREST_SIZE_LIMIT: int = 100
+_MINING_PAGE_LIMIT: int = 200
+_TRADE_RECORD_LIMIT: int = 1000
+_WITHDRAWAL_RECORD_LIMIT: int = 1000
+
+# Types of Binance Dividends
+_BNB_VAULT = "BNB Vault"
+_ETH_STAKING = "ETH 2.0 Staking"
+_FLEXIBLE = "Flexible"
+_FLEXIBLE_SAVINGS = "Flexible Savings"
+_LAUNCH_POOL = "Launchpool"
+_LOCKED = "Locked"
+_LOCKED_SAVINGS = "Locked Savings"
+_LOCKED_STAKING = "Locked Staking"
+_SOLO_AIRDROP = "SOLO airdrop"
+_GENERAL_STAKING = "STAKING"
+# More types added
+_CASH_VOUCHER = "Cash Voucher"
+_LAUNCHPAD = "launchpad"
+_SAVINGS_TRAIL_FUND = "Savings Trail Fund"
+
+_AIRDROP_LIST = [_SOLO_AIRDROP]
+_INTEREST_LIST = [_FLEXIBLE, _FLEXIBLE_SAVINGS, _LOCKED, _LOCKED_SAVINGS, _SAVINGS_TRAIL_FUND]
+_STAKING_LIST = [_ETH_STAKING, _LOCKED_STAKING, _BNB_VAULT, _LAUNCH_POOL, _GENERAL_STAKING, _LAUNCHPAD]
+_INCOME_LIST = [_CASH_VOUCHER]
+
+
+class InputPlugin(AbstractCcxtInputPlugin):
+    __EXCHANGE_NAME: str = "Binance.us"
+    __PLUGIN_NAME: str = "Binance.us_REST"
+    __DEFAULT_THREAD_COUNT: int = 1
+
+    def __init__(
+        self,
+        account_holder: str,
+        api_key: str,
+        api_secret: str,
+        native_fiat: str,
+        username: Optional[str] = None,
+        thread_count: Optional[int] = __DEFAULT_THREAD_COUNT,
+    ) -> None:
+        self.__api_key = api_key
+        self.__api_secret = api_secret
+        # We will have a default start time of July 13th, 2017 since Binance Exchange officially launched on July 14th Beijing Time.
+        super().__init__(account_holder, datetime(2017, 7, 13, 0, 0, 0, 0), native_fiat, thread_count)
```

Comment:
Can you track down the start date for Binance.us?

Reply:
How exact do I need to be? I think it is 09/22/2019 to the nearest day.

Reply:
An safe approximation is okay. Err on the side of being too early. For example, if you think it might be 9/22/2019 EST, you might as well make it 9/21/2019 just to be safe.

This just helps with optimization, so that we don't pull unnecessary information.